services/dask-sidecar/src/simcore_service_dask_sidecar/utils/files.py

```diff
@@ -140,6 +141,22 @@ async def _copy_file(
 _ZIP_MIME_TYPE: Final[str] = "application/zip"
 
 
+def check_need_unzipping(
+    src_url: AnyUrl,
+    target_mime_type: str | None,
+    dst_path: Path,
+) -> bool:
+    """
+    Checks if the source URL points to a zip file and if the target mime type is not zip.
+    If so, extraction is needed.
+    """
+    src_mime_type, _ = mimetypes.guess_type(f"{src_url.path}")
+    dst_mime_type = target_mime_type
+    if not dst_mime_type:
+        dst_mime_type, _ = mimetypes.guess_type(dst_path)
+    return (src_mime_type == _ZIP_MIME_TYPE) and (dst_mime_type != _ZIP_MIME_TYPE)
```

Comment:
i see. so the source is zipped and the destination expects something that is unzipped. In that case, you need to unzip first

what about `check_zipped_source_to_unzipped_dest`

Reply:
ok I agree I should uniformise the names. but I will not write something like this. it's long while still using incomplete names ;)